projects/packages/connection/src/class-manager.php

```diff
@@ -972,6 +972,20 @@ public function register( $api_endpoint = 'register' ) {
 
 		$this->get_tokens()->update_blog_token( (string) $registration_details->jetpack_secret );
 
+		if ( empty( $registration_details->allow_inplace_authorization ) ) {
+			// Forces register_site REST endpoint to return the Calypso authorization URL.
+			add_filter( 'jetpack_use_iframe_authorization_flow', '__return_false', 20 );
+		}
+
+		add_filter(
+			'jetpack_register_site_rest_response',
+			function ( $response ) use ( $registration_details ) {
+				$response['allowInplaceAuthorization'] = $registration_details->allow_inplace_authorization;
+				$response['alternateAuthorizeUrl']     = $registration_details->alternate_authorization_url;
```

Comment:
I wonder if we should add some kind of validation here? Maybe something that limits things to a list of domains we own. Would that be worth it?

Reply:
Good point. Since this is added via a filter, any plugin would be able to hijack this and send the user somewhere else.

I'd say we could pass a string with a redirect `source` and always use our Redirect mechanism, this would keep us safe, however, we might also want to redirect the user somewhere in the wp-admin dashboard.

Thinking about 2 options:

1. allow only redirect slugs and URLs inside wp-admin
2. allow only redirect slugs and routes inside the Jetpack React dashboard.

Reply:
I don't think it's a good idea to allow this URL to be filtered, a plugin could disrupt our onboarding flow.

So I decided to restrict it only to our Redirect service, this gives us control over this URL and already accepts only URLs from domains we own.

https://github.com/Automattic/jetpack/pull/19833/commits/951cc893090aaf142fbb7dc145401b42a46f7c2a